addons/stock/wizard/stock_replenishment_info.py

```diff
@@ -73,38 +95,98 @@ def _format_description(description):
+                curve_line_vals.append({'x': date_string, 'y': self.product_min_qty})
+                curve_line_vals.append({'x': date_string, 'y': self.product_max_qty})
+            curve_line_vals.pop()   # we pop the last value since it would result in an ascending line that we don't need
+
+        max_line_vals = [{'x': date, 'y': self.product_max_qty} for date in axe_x_vals]
+        min_line_vals = [{'x': date, 'y': self.product_min_qty} for date in axe_x_vals]
+        graph_data = {
+            'axe_x_vals': axe_x_vals,
+            'max_line_vals': max_line_vals,
+            'min_line_vals': min_line_vals,
+            'curve_line_vals': curve_line_vals,
+        }
+        return ordering_period, graph_data
+
+    @api.depends('orderpoint_id', 'based_on', 'percent_factor', 'product_min_qty', 'product_max_qty')
+    def _compute_json_replenishment_graph(self):
+        for replenishment_report in self:
+            if not replenishment_report.product_id or not replenishment_report.orderpoint_id.location_id:
+                continue
+            lead_days, _ = replenishment_report._get_lead_days_and_description()
+            date_from, date_to = replenishment_report._get_period_of_time()
             domain = Domain([
                 ('product_id', '=', replenishment_report.product_id.id),
                 ('date', '>=', date_from),
                 ('date', '<=', datetime.combine(date_to, time.max)),
                 ('state', '=', 'done'),
                 ('company_id', '=', replenishment_report.orderpoint_id.company_id.id)
             ])
-            quantity_by_month_out = self.env['stock.move']._read_group(
-                domain & Domain('location_dest_id.usage', '=', 'customer'),
-                ['date:month'], ['product_qty:sum'])
-            quantity_by_month_returned = dict(self.env['stock.move']._read_group(
-                domain & Domain('location_id.usage', '=', 'customer'),
-                ['date:month'], ['product_qty:sum']))
-            locale = get_lang(self.env).code
-            fmt = models.READ_GROUP_DISPLAY_FORMAT['month']
-            for month, product_qty_sum in quantity_by_month_out:
-                replenishment_history.append({
-                    'name': babel.dates.format_datetime(month, format=fmt, locale=locale),
-                    'quantity': product_qty_sum - quantity_by_month_returned.get(month, 0),
-                    'uom_name': replenishment_report.product_id.uom_id.display_name,
-                })
-            replenishment_report.json_replenishment_history = dumps({
-                'template': 'stock.replenishmentHistory',
-                'replenishment_history': replenishment_history
+            quantity_out = sum(self.env['stock.move'].search(domain & Domain('location_dest_id.usage', '=', 'customer')).mapped('product_qty'))
+            quantity_returned = sum(self.env['stock.move'].search(domain & Domain('location_id.usage', '=', 'customer')).mapped('product_qty'))
```

Comment:
Should probably still use `_read_group` with a `:sum` aggregate to avoid fully loading the move records into memory unnecessarily